### PR TITLE
Terraform aws provider v4.0.0 updates

### PR DESCRIPTION
**Pull request by NikitaGandhi**
**Any PRs will require running `terraform fmt -recursive` successfully first. Please install terraform version > `v0.15` on your local setup for this activity.**

# Why this change is needed
> Describe why this change is needed, what issues it will fix and the benefits the change will add

The recent release of `terraform-aws-provider` `v4.0.0` has remove support for some of the `aws_s3_bucket` resource configurations. This PR update/fix those changes. Refer release [v4.0.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v4.0.0) for more detail.

# Negative effects of this change
> Will making this change break or change an existing functionality? flag it here

Reapply bucket configurations.

### CODE REVIEW

**Comment by jmettes** [comment]
The Terraform upgrade guide might be useful to other reviewers to refer to:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/version-4-upgrade

**Review by pindge — APPROVED** [review]

**Review by jmettes — CHANGES_REQUESTED** [review]
Some minor things I noticed

File: odc_eks/cloudfront_distribution.tf [hunk 1]
```diff
@@ -201,6 +191,51 @@ resource "aws_s3_bucket" "cloudfront_log_bucket" {
   )
 }
 
+resource "aws_s3_bucket_acl" "cloudfront_log_bucket" {
+  count  = (var.cf_log_bucket_create && var.cf_enable) ? 1 : 0
+  bucket = aws_s3_bucket.cloudfront_log_bucket[0].id
+
+  access_control_policy {
```
[the remark this answers is not in the document]
Comment (NikitaGandhi) [reply odc_eks/cloudfront_distribution.tf:198]: you are only allow to set `acl` or `access_control_policy`

[the remark this answers is not in the document]
Comment (jmettes) [reply odc_eks/cloudfront_distribution.tf:198]: Yeah - it just seems like it's been given more access than it previously had?

The other two acls in this PR have, whereas this one doesn't. Just checking whether that's the intention.

```
resource "aws_s3_bucket_acl" ... {
   ...
  acl = "private" 
}
```

The original bucket resource didn't have any grant statements.

**Review by jmettes — CHANGES_REQUESTED** [review]

File: odc_eks/cloudfront_distribution.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmettes) [reply odc_eks/cloudfront_distribution.tf:198]: Yeah - it just seems like it's been given more access than it previously had?

The other two acls in this PR use `acl="private"`, whereas this one doesn't. Just checking whether that's the intention.

```
resource "aws_s3_bucket_acl" ... {
   ...
  acl = "private" 
}
```

The original bucket resource didn't have any grant statements.

[the remark this answers is not in the document]
Comment (NikitaGandhi) [reply odc_eks/cloudfront_distribution.tf:198]: Yo needed ACL permission for cloudfront awslogdelivery account which was missing before (actually s3 ACL permission got update during cloudfront provisioning. you can review your own cloudfront log bucket and confirm the same). This change will make sure the permission remain same.

Also refer this doc for cloudfront logging doc: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AccessLogs.html

**Review by jmettes — APPROVED** [review]

File: odc_eks/cloudfront_distribution.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmettes) [reply odc_eks/cloudfront_distribution.tf:198]: Ah thanks! I understand now

**Comment by jmettes** [comment]
Looks good! I'll update the pinned v3 providers after this is merged.

**Pull request merged**